src/test/java/org/entando/plugins/pda/controller/ConnectionsControllerIntegrationTest.java

```diff
@@ -0,0 +1,274 @@
+import org.entando.plugins.pda.dto.connection.ConnectionDto;
+import org.entando.plugins.pda.pam.engine.KieEngine;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.cloud.contract.wiremock.AutoConfigureWireMock;
+import org.springframework.test.context.ActiveProfiles;
+import org.springframework.test.context.TestExecutionListeners;
+import org.springframework.test.context.junit4.SpringRunner;
+import org.springframework.test.context.support.DependencyInjectionTestExecutionListener;
+import org.springframework.test.web.servlet.MockMvc;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import static com.github.tomakehurst.wiremock.client.WireMock.*;
+import static org.hamcrest.Matchers.hasSize;
+import static org.hamcrest.Matchers.is;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.delete;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.put;
+import static org.springframework.test.web.servlet.result.MockMvcResultHandlers.print;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.jsonPath;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+@AutoConfigureMockMvc
+@ActiveProfiles("test")
+@RunWith(SpringRunner.class)
+@TestExecutionListeners({ DependencyInjectionTestExecutionListener.class })
+@AutoConfigureWireMock(port = 8089)
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+public class ConnectionsControllerIntegrationTest {
+
+    @Autowired private MockMvc mockMvc;
+
+    private ObjectMapper mapper = new ObjectMapper();
+
+    @Before
+    public void setup() {
+        stubFor(WireMock.get(urlEqualTo("/config/entando-process-driven-plugin"))
+            .willReturn(aResponse().withStatus(200).withHeader("Content-Type", "application/json")
+                .withBodyFile("mock_connection_configs.json")));
+
+        stubFor(WireMock.put(urlEqualTo("/config/entando-process-driven-plugin"))
+                .willReturn(aResponse().withStatus(200)));
+    }
+
+    @Test
+    public void testListConnections() throws Exception {
+        mockMvc.perform(get("/connections"))
+                .andDo(print()).andExpect(status().isOk())
+                .andExpect(jsonPath("errors", hasSize(0)))
+                .andExpect(jsonPath("payload.size()", is(2)))
+                .andExpect(jsonPath("payload[0].name", is("kieProduction")))
+                .andExpect(jsonPath("payload[1].name", is("kieStaging")))
+                ;
```

Comment:
Strange placement for a semicolon. I think it can go on the previous line, like you are doing for the builder. The same for the other cases.